# [PATCH] remove icq integration
# Author: yakov

--- a/usercp.php
+++ b/usercp.php
@@ -228,7 +228,7 @@ if($mybb->input['action'] == "do_profile" && $mybb->request_method == "post")
 		"away" => $away,
 		"profile_fields" => $mybb->get_input('profile_fields', MyBB::INPUT_ARRAY)
 	));
-	foreach(array('icq', 'skype', 'google') as $cfield)
+	foreach(array('skype', 'google') as $cfield)
 	{
 		$csetting = 'allow'.$cfield.'field';
 		if($mybb->settings[$csetting] == '')
@@ -241,23 +241,11 @@ if($mybb->input['action'] == "do_profile" && $mybb->request_method == "post")
 			continue;
 		}
 
-		if($cfield == 'icq')
-		{
-			$user[$cfield] = $mybb->get_input($cfield, 1);
+		$user[$cfield] = $mybb->get_input($cfield);
 
-			if(my_strlen($user[$cfield]) > 10)
-			{
-				error($lang->contact_field_icqerror);
-			}
-		}
-		else
+		if(my_strlen($user[$cfield]) > 75)
 		{
-			$user[$cfield] = $mybb->get_input($cfield);
-
-			if(my_strlen($user[$cfield]) > 75)
-			{
-				error($lang->contact_field_error);
-			}
+			error($lang->contact_field_error);
 		}
 	}
 
@@ -379,16 +367,6 @@ if($mybb->input['action'] == "profile")
 		$user['website'] = htmlspecialchars_uni($user['website']);
 	}
 
-	if($user['icq'] != "0")
-	{
-		$user['icq'] = (int)$user['icq'];
-	}
-
-	if($user['icq'] == 0)
-	{
-		$user['icq'] = '';
-	}
-
 	if($errors)
 	{
 		$user['skype'] = htmlspecialchars_uni($user['skype']);
@@ -399,7 +377,7 @@ if($mybb->input['action'] == "profile")
 	$contactfields = '';
 	$cfieldsshow = false;
 
-	foreach(array('icq', 'skype', 'google') as $cfield)
+	foreach(array('skype', 'google') as $cfield)
 	{
 		$contact_fields[$cfield] = '';
 		$csetting = 'allow'.$cfield.'field';

--- a/modcp.php
+++ b/modcp.php
@@ -2569,7 +2569,6 @@ if($mybb->input['action'] == "do_editprofile")
 		"profile_fields" => $mybb->get_input('profile_fields', MyBB::INPUT_ARRAY),
 		"profile_fields_editable" => true,
 		"website" => $mybb->get_input('website'),
-		"icq" => $mybb->get_input('icq'),
 		"skype" => $mybb->get_input('skype'),
 		"google" => $mybb->get_input('google'),
 		"signature" => $mybb->get_input('signature'),
@@ -2778,11 +2777,6 @@ if($mybb->input['action'] == "editprofile")
 		$user['website'] = '';
 	}
 
-	if($user['icq'] != "0")
-	{
-		$user['icq'] = (int)$user['icq'];
-	}
-
 	if(!$errors)
 	{
 		$mybb->input = array_merge($user, $mybb->input);
@@ -2803,7 +2797,7 @@ if($mybb->input['action'] == "editprofile")
 	}
 
 	// Sanitize all input
-	foreach(array('usertitle', 'website', 'icq', 'skype', 'google', 'signature', 'birthday_day', 'birthday_month', 'birthday_year') as $field)
+	foreach(array('usertitle', 'website', 'skype', 'google', 'signature', 'birthday_day', 'birthday_month', 'birthday_year') as $field)
 	{
 		$mybb->input[$field] = htmlspecialchars_uni($mybb->get_input($field));
 	}
@@ -3273,7 +3267,6 @@ if($mybb->input['action'] == "editprofile")
 
 	$birthday_year = $mybb->input['birthday_year'];
 	$user_website = $mybb->input['website'];
-	$user_icq = $mybb->input['icq'];
 	$user_skype = $mybb->input['skype'];
 	$user_google = $mybb->input['google'];
 
--- a/memberlist.php
+++ b/memberlist.php
@@ -67,7 +67,7 @@ if($mybb->get_input('action') == "search")
 	}
 
 	$contact_fields = array();
-	foreach(array('skype', 'google', 'icq') as $field)
+	foreach(array('skype', 'google') as $field)
 	{
 		$contact_fields[$field] = '';
 		$settingkey = 'allow'.$field.'field';
@@ -257,7 +257,7 @@ else
 	}
 
 	// Search by contact field input
-	foreach(array('icq', 'google', 'skype') as $cfield)
+	foreach(array('google', 'skype') as $cfield)
 	{
 		$csetting = 'allow'.$cfield.'field';
 		$mybb->input[$cfield] = trim($mybb->get_input($cfield));
@@ -287,14 +287,8 @@ else
 				}
 				$search_query .= ")";
 			}
-			if($cfield == 'icq')
-			{
-				$search_query .= " AND u.{$cfield} LIKE '%".(int)$mybb->input[$cfield]."%'";
-			}
-			else
-			{
-				$search_query .= " AND u.{$cfield} {$like} '%".$db->escape_string_like($mybb->input[$cfield])."%'";
-			}
+			
+			$search_query .= " AND u.{$cfield} {$like} '%".$db->escape_string_like($mybb->input[$cfield])."%'";
 			$search_url .= "&{$cfield}=".urlencode($mybb->input[$cfield]);
 		}
 	}

--- a/member.php
+++ b/member.php
@@ -2105,7 +2105,7 @@ if($mybb->input['action'] == "profile")
 
 	$contact_fields = array();
 	$any_contact_field = false;
-	foreach(array('icq', 'skype', 'google') as $field)
+	foreach(array('skype', 'google') as $field)
 	{
 		$contact_fields[$field] = '';
 		$settingkey = 'allow'.$field.'field';
@@ -2113,15 +2113,7 @@ if($mybb->input['action'] == "profile")
 		if(!empty($memprofile[$field]) && is_member($mybb->settings[$settingkey], array('usergroup' => $memprofile['usergroup'], 'additionalgroups' => $memprofile['additionalgroups'])))
 		{
 			$any_contact_field = true;
-
-			if($field == 'icq')
-			{
-				$memprofile[$field] = (int)$memprofile[$field];
-			}
-			else
-			{
-				$memprofile[$field] = htmlspecialchars_uni($memprofile[$field]);
-			}
+			$memprofile[$field] = htmlspecialchars_uni($memprofile[$field]);
 			$tmpl = 'member_profile_contact_fields_'.$field;
 
 			$bgcolors[$field] = alt_trow();

--- a/inc/datahandlers/user.php
+++ b/inc/datahandlers/user.php
@@ -317,24 +317,6 @@ class UserDataHandler extends DataHandler
 		return true;
 	}
 
-	/**
-	 * Verifies if an ICQ number is valid or not.
-	 *
-	 * @return boolean True when valid, false when invalid.
-	 */
-	function verify_icq()
-	{
-		$icq = &$this->data['icq'];
-
-		if($icq != '' && !is_numeric($icq))
-		{
-			$this->set_error("invalid_icq_number");
-			return false;
-		}
-		$icq = (int)$icq;
-		return true;
-	}
-
 	/**
 	* Verifies if a birthday is valid or not.
 	*
@@ -1017,10 +999,6 @@ class UserDataHandler extends DataHandler
 		{
 			$this->verify_website();
 		}
-		if($this->method == "insert" || array_key_exists('icq', $user))
-		{
-			$this->verify_icq();
-		}
 		if($this->method == "insert" || (isset($user['birthday']) && is_array($user['birthday'])))
 		{
 			$this->verify_birthday();
@@ -1121,7 +1099,7 @@ class UserDataHandler extends DataHandler
 
 		$user = &$this->data;
 
-		$array = array('postnum', 'threadnum', 'avatar', 'avatartype', 'additionalgroups', 'displaygroup', 'icq', 'skype', 'google', 'bday', 'signature', 'style', 'dateformat', 'timeformat', 'notepad', 'regip', 'lastip', 'coppa_user');
+		$array = array('postnum', 'threadnum', 'avatar', 'avatartype', 'additionalgroups', 'displaygroup', 'skype', 'google', 'bday', 'signature', 'style', 'dateformat', 'timeformat', 'notepad', 'regip', 'lastip', 'coppa_user');
 		foreach($array as $value)
 		{
 			if(!isset($user[$value]))
@@ -1163,7 +1141,6 @@ class UserDataHandler extends DataHandler
 			"lastactive" => (int)$user['lastactive'],
 			"lastvisit" => (int)$user['lastvisit'],
 			"website" => $db->escape_string($user['website']),
-			"icq" => (int)$user['icq'],
 			"skype" => $db->escape_string($user['skype']),
 			"google" => $db->escape_string($user['google']),
 			"birthday" => $user['bday'],
@@ -1373,10 +1350,6 @@ class UserDataHandler extends DataHandler
 		{
 			$this->user_update_data['website'] = $db->escape_string($user['website']);
 		}
-		if(isset($user['icq']))
-		{
-			$this->user_update_data['icq'] = (int)$user['icq'];
-		}
 		if(isset($user['skype']))
 		{
 			$this->user_update_data['skype'] = $db->escape_string($user['skype']);
@@ -1788,7 +1761,6 @@ class UserDataHandler extends DataHandler
 		$update = array(
 			"website" => "",
 			"birthday" => "",
-			"icq" => "",
 			"skype" => "",
 			"google" => "",
 			"usertitle" => "",

--- a/admin/modules/user/users.php
+++ b/admin/modules/user/users.php
@@ -519,7 +519,6 @@ if($mybb->input['action'] == "edit")
 			"profile_fields" => $mybb->get_input('profile_fields', MyBB::INPUT_ARRAY),
 			"profile_fields_editable" => true,
 			"website" => $mybb->get_input('website'),
-			"icq" => $mybb->get_input('icq'),
 			"skype" => $mybb->get_input('skype'),
 			"google" => $mybb->get_input('google'),
 			"birthday" => array(
@@ -1175,7 +1174,6 @@ EOF;
 	$form_container = new FormContainer($lang->optional_profile_info.': '.htmlspecialchars_uni($user['username']));
 	$form_container->output_row($lang->custom_user_title, $lang->custom_user_title_desc, $form->generate_text_box('usertitle', $mybb->get_input('usertitle'), array('id' => 'usertitle')), 'usertitle');
 	$form_container->output_row($lang->website, "", $form->generate_text_box('website', $mybb->get_input('website'), array('id' => 'website')), 'website');
-	$form_container->output_row($lang->icq_number, "", $form->generate_numeric_field('icq', $mybb->get_input('icq'), array('id' => 'icq', 'min' => 0)), 'icq');
 	$form_container->output_row($lang->skype_handle, "", $form->generate_text_box('skype', $mybb->get_input('skype'), array('id' => 'skype')), 'skype');
 	$form_container->output_row($lang->google_handle, "", $form->generate_text_box('google', $mybb->get_input('google'), array('id' => 'google')), 'google');
 
